### DISCUSSION TITLE
How to limit scroll area height?

### DISCUSSION THREAD

**Discussion by 90-008** [emilk/egui#989] [Q&A]
When i'm using a scroll area with `.auto_shrink([false, false])`, the scroll area covers all of the available height, leaving no space for widgets added to the bottom of it. How can i make it so that the bottom widgets limit the height of the scroll area?

**Comment by parasyte** [comment]
You will probably have to put the ScrollArea inside a container. I’ve used it in a `CentralPanel` with other controls in a `TopBottomPanel`.

I believe the `ScrollArea` respects the size of other widgets when they request a minimum size from the layout allocator.

**Reply by 90-008** [reply]
I also have it in a `CentralPanel`, like so:
```rust
        egui::CentralPanel::default().show(ctx, |ui| {
            ui.with_layout(
                Layout::from_main_dir_and_cross_align(egui::Direction::LeftToRight, egui::Align::Center),
                |ui| {
                    ui.vertical(|ui| {
                        self.view_messages(state, ui);
                        ui.separator();
                        self.view_composer(state, ui, ctx);
                    });
                },
            );
        });
```
`view_messages` is a method that creates a `ScrollArea`, and `view_composer` is a method that creates a multiline text input. What I want to have is the `ScrollArea` should get smaller if the text input gets larger, eg. more lines are created, but instead what happens is the text input overflows the bottom of the screen.
